### PR TITLE
Add more const correctness

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
Travis Is Ok for me.

### CODE REVIEW

File: crypto/x509v3/v3_ia5.c [hunk 1]
```diff
@@ -90,7 +90,7 @@ char *i2s_ASN1_IA5STRING(X509V3_EXT_METHOD *method, ASN1_IA5STRING *ia5)
 }
 
 ASN1_IA5STRING *s2i_ASN1_IA5STRING(X509V3_EXT_METHOD *method,
-                                   X509V3_CTX *ctx, char *str)
+                                   X509V3_CTX *ctx, const char *str)
```
Comment (mattcaswell) [inline crypto/x509v3/v3_ia5.c]: The cast of `str` inside this function looks a bit odd and casts away the `const`...although its ok really.

Comment (FdaSilvaYY) [inline crypto/x509v3/v3_ia5.c]: You're right, the cast is useless.
The second argument is typed 'const void* ' ...  and it is directly recast to 'const char *' on the other side.
I removed it.

**Comment by mattcaswell** [comment]
+1

**Comment by kroeckx** [comment]
+1

File: crypto/asn1/asn1_gen.c [hunk 2]
```diff
@@ -86,7 +86,7 @@ ASN1_TYPE *ASN1_generate_nconf(char *str, CONF *nconf)
     return ASN1_generate_v3(str, &cnf);
 }
 
-ASN1_TYPE *ASN1_generate_v3(char *str, X509V3_CTX *cnf)
+ASN1_TYPE *ASN1_generate_v3(const char *str, X509V3_CTX *cnf)
```
Comment (richsalz) [inline crypto/asn1/asn1_gen.c]: doc update.

File: crypto/x509/x509_lu.c [hunk 3]
```diff
@@ -370,7 +370,7 @@ int X509_OBJECT_up_ref_count(X509_OBJECT *a)
     return 1;
 }
 
-X509 *X509_OBJECT_get0_X509(X509_OBJECT *a)
+X509 *X509_OBJECT_get0_X509(const X509_OBJECT *a)
```
Comment (richsalz) [inline crypto/x509/x509_lu.c]: doc update?

**Comment by richsalz** [comment]
Nice, but these need to be checked for doc updates.

File: crypto/asn1/asn1_gen.c (hunk 2 above)
Comment (FdaSilvaYY) [inline crypto/asn1/asn1_gen.c]: Hi,
It's already in this ¨R, see ASN1_generate_nconf.pod below

File: crypto/x509/x509_lu.c (hunk 3 above)
Comment (FdaSilvaYY) [inline crypto/x509/x509_lu.c]: I have found any doc about X509_OBJECT type and his related methods.
It's may be considered as an internal data type ?

**Comment by richsalz** [comment]
I missed that you did update the pod page for one, sorry.  No need to write new docs, but every that exists should be updated if needed.  If you say you did that, great.

**Comment by FdaSilvaYY** [comment]
AFAIK, I keep sync the method declarations in code, and docs.
But most of them are not documented, that why you see so few pod changes.

**Comment by richsalz** [comment]
Okay, I will merge.

--
Senior Architect, Akamai Technologies
IM: richsalz@jabber.at Twitter: RichSalz

**Comment by FdaSilvaYY** [comment]
Merged as :
7d7da288b81aea19622ea1820722ce869fbe0364
2b91da968cec55de9d6db464c8c938d2930832e9
13f74c66ce83cc554ed29f88706a3176a1788f45
12eaf3b8496f5e3578414202f7d26dc5cec2ee9a
009951d24da57089072949f84acc5c906b15fe71
3470795171b8aecd4dbc86061600093b86b23c0e

Thanks.

**Pull request closed without merging**